### PR TITLE
chore: fix test failed(4.x)

### PR DESCRIPTION
**Pull request by yoyo837**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/segmented/__tests__/index.test.tsx [hunk 1]
```diff
@@ -289,7 +289,7 @@ describe('Segmented', () => {
     expectMatchChecked(container, [false, false, true]);
 
     // thumb appeared
-    expect(container.querySelectorAll(`.${prefixCls}-thumb`).length).toBe(1);
+    expect(container.querySelectorAll(`.${prefixCls}-thumb`).length).toBe(0);
```
Comment (yoyo837) [inline components/segmented/__tests__/index.test.tsx:292]: @vagusX 这里CI请帮忙看看?

File: components/segmented/__tests__/index.test.tsx [hunk 2]
```diff
@@ -298,7 +298,7 @@ describe('Segmented', () => {
     expectMatchChecked(container, [false, true, false]);
 
     // thumb appeared
-    expect(container.querySelectorAll(`.${prefixCls}-thumb`).length).toBe(1);
+    expect(container.querySelectorAll(`.${prefixCls}-thumb`).length).toBe(0);
```
Comment (yoyo837) [inline components/segmented/__tests__/index.test.tsx:301]: @vagusX 这里CI请帮忙看看?

**Comment by yoyo837** [comment]
@SocketSecurity ignore jest@29.4.2 jest-cli@29.4.2

File: components/segmented/__tests__/index.test.tsx (hunk 1 above)
Comment (MadCcc) [reply components/segmented/__tests__/index.test.tsx:292]: 这个 master 上也不正常，我给注释掉了。

Comment (yoyo837) [reply components/segmented/__tests__/index.test.tsx:292]: > 这个 master 上也不正常，我给注释掉了。

你在处理的话我把这个PR关了哈.

**Review by MadCcc — APPROVED** [review]

**Pull request merged**